src/EditorFeatures/Core/Implementation/NavigateTo/NavigateToSearchResultProvider.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.Navigation;
+using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Microsoft.VisualStudio.Language.NavigateTo.Interfaces;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.NavigateTo
+{
+    [Export(typeof(INavigateToSearchResultProvider)), Shared]
```

Comment:
TS has its own navigate to search service http://index/#Microsoft.CodeAnalysis.TypeScript.EditorFeatures/Features/NavigateTo/TypeScriptNavigateToSearchService.cs,12, which is not based on providers. We probably need the approach that HeeJae recommends as the provider probably needs to be able to register for specific workspace(s) only.